### PR TITLE
Pallet updater voting

### PR DESCRIPTION
**Pull request by zycon91**
add: 
- [x] storage items for Proposals, Voting and ProposalCount.
- [x] struct for votes.
- [x] propose function.
- [x] vote function.
- [x] close_vote function.

### CODE REVIEW

File: bin/node/pallets/pallet-updater/src/lib.rs [hunk 1]
```diff
@@ -106,7 +156,73 @@ pub mod pallet {
...
+			Self::deposit_event(Event::RemovedUpdater(remove_acct));
+
+			Ok(())
+		}
+
+		/// Adds a proposal to be voted on.
+		///
+		/// Requires the sender to be a member.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn propose_code(origin: OriginFor<T>, code: Vec<u8>) -> DispatchResult {
```
Comment (charmitro) [inline bin/node/pallets/pallet-updater/src/lib.rs:168]: What is this? What does `code: Vec<u8>` means in our context?

**Review by charmitro — CHANGES_REQUESTED** [review]

File: bin/node/pallets/pallet-updater/src/lib.rs (hunk 1 above)
Comment (zycon91) [reply bin/node/pallets/pallet-updater/src/lib.rs:168]: > What is this? What does `code: Vec<u8>` means in our context?

I thought that we could give as input the code of the upgrade (like in the set_code function).

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -233,11 +233,12 @@ impl pallet_randomness_collective_flip::Config for Runtime {}
 parameter_types! {
 	/// Number of maximum members.
 	pub const MaxMemberCnt: u32 = 3;
+	pub const MaxProposalCnt: u32 = 5;
```
[the remark this answers is not in the document]
Comment (zycon91) [reply bin/node/runtime/src/lib.rs:236]: > Probably MaxProposalCnt should be `1` if we are going to use this pallet for only runtime upgrades.
> 
> We should not allow more than one runtime upgrade proposal.

Ok. Sounds like a plan!

File: bin/node/pallets/pallet-updater/src/functions.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+        Ok(())
+    }
+
+    pub fn update_code_in_storage(code: &[u8]) -> DispatchResult {
+        // Write code to the storage and emit related events and digest items.
+        
+        Ok(())
+    }
+}
```
Comment (charmitro) [inline bin/node/pallets/pallet-updater/src/functions.rs:21]: We are probably going to use `frame/system`. Remove these and we'll introduce them later if needed.

**Review by charmitro — CHANGES_REQUESTED** [review]

File: bin/node/pallets/pallet-updater/src/functions.rs (hunk 3 above)
Comment (zycon91) [reply bin/node/pallets/pallet-updater/src/functions.rs:21]: > We are probably going to use `frame/system`. Remove these and we'll introduce them later if needed.

Ok. Sounds good.

**Review by charmitro — APPROVED** [review]

**Pull request merged**